exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -391,7 +397,10 @@ private void parseAndRunPhysicalPlan(final String json) throws ExecutionSetupExc
   private void runPhysicalPlan(final PhysicalPlan plan) throws ExecutionSetupException {
     validatePlan(plan);
     setupSortMemoryAllocations(plan);
-    acquireQuerySemaphore(plan);
+    if (queuingEnabled) {
+      acquireQuerySemaphore(plan);
+    }
+    recordNewState(QueryState.STARTING);
```

Comment:
If the initialState is STARTING won't this fail (state transition from STARTING to STARTING is invalid given the fall-through)? The recordNewState(STARTING) should be inside the if condition, right?